### PR TITLE
seastar: bump up seastar submodule to recent master

### PR DESCRIPTION
**Pull request by tchaikov**
to address the fmtlib v9 compatibility issues.

Signed-off-by: Kefu Chai <tchaikov@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-fmt-v9/924a046eaa085c0ee72c841ba2ae4af3abe27410/

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-fmt-v9-4/58bc30fd33deff91a1e3c670841346a08b1fa482/

**Comment by tchaikov** [comment]
jenkins test api

**Comment by tchaikov** [comment]
jenkins test make check

**Review by cyx1231st — APPROVED** [review]
@athanatos  looks that we have to customize seastar now to make the errorator work: https://github.com/ceph/seastar/commit/b85ba6209796e53729499879a7ee5bcf17aa82b1

File: src/crimson/os/seastore/onode_manager/staged-fltree/fwd.h [hunk 1]
```diff
@@ -189,3 +189,10 @@ void reset_ptr(PtrType& ptr, const char* origin_base,
 }
 
 }
+
+#if FMT_VERSION >= 90000
+template<>
+struct fmt::formatter<crimson::os::seastore::onode::tree_stats_t> : fmt::ostream_formatter {};
+template<>
+struct fmt::formatter<rados::cls::fifo::part_header> : fmt::ostream_formatter {};
```
Comment (cyx1231st) [inline src/crimson/os/seastore/onode_manager/staged-fltree/fwd.h:197]: Hmm, why need this?

Comment (tchaikov) [reply src/crimson/os/seastore/onode_manager/staged-fltree/fwd.h:197]: will drop it.

Comment (tchaikov) [reply src/crimson/os/seastore/onode_manager/staged-fltree/fwd.h]: @cyx1231st dropped.

**Pull request merged**